curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md

```diff
@@ -0,0 +1,157 @@
+---
+id: 66db529d37ad966480ebb633
+title: Build a Lunch Picker Program
+challengeType: 14
+dashedName: build-a-lunch-picker-program
+---
+
+# --description--
+
+In this lab, you'll build a program to assist in choosing lunch options. You'll work with two arrays: one for children's food choices and one for adults. You'll add and remove items from these arrays and then randomly select a meal from each. Finally, you'll log the results to the console.
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. You should create a `kidsMeal` variable and assign it an array with at least three strings in it.
+2. You should create an `adultsMeal` variable and assign it an array with at least three strings in it.
+3. Once items have been added to the `kidsMeal`, you should use the appropriate array method to remove the last item.
+4. Once items have been added to the `adultsMeal`, you should use the appropriate array method to remove the first item. 
+5.  You should use the appropriate `Math` methods to generate a random number between `0` to the length of the `kidsMeal` array. Assign that result to a variable called `randomIndexKids`.
+6. You should use the appropriate `Math` methods to generate a random number between `0` to the length of the `kidsMeal` array. Assign that result to a variable called `randomIndexAdults`.
+7.  Log the following messages to the console:
+
+- `"Today kids should eat:"` followed by the selected meal to the console.
+- `"Today adults should eat:"` followed by the selected meal to the console.
+
+# --hints--
+
+You should declare a `kidsMeal` array.
+
+```js
+assert.isArray(kidsMeal);
+```
+
+You should declare an `adultsMeal` array.
+
+```js
+assert.isArray(adultsMeal);
+```
+
+Your `kidsMeal` array should contain atleast three items.
+
+```js
+const matched = __helpers.removeJSComments(code).match(/(const|let)\s+kidsMeal\s*=\s*\[(?<items>.+?)\]/)
+const kidsMealArray = matched.groups.items.split(",");
+assert.isAtLeast(kidsMealArray.length, 3);
+```
+
+Your `kidsMeal` array should contain only non-empty string values.
+
+```js
+const matched = __helpers.removeJSComments(code).match(/(const|let)\s+adultsMeal\s*=\s*\[(?<items>.+?)\]/)
+const adultsMealArray = matched.groups.items.split(",");
+assert.isAtLeast(adultsMealArray.length, 1); 
+
+for (let i = 0; i < kidsMeal.length; i++) {
+  assert.isString(kidsMeal[i]); // Check that every item is a string
+  assert.isAbove(kidsMeal[i].length, 0, 'String should not be empty'); // Check that string is not empty
+}
```

Comment:
The test is broken (referencing `adultsMeal`) but it doesn't work because the match is always a string. The following should work well.
The problem is if the items are variables. For that we need a different solution.
```suggestion
const matched = __helpers.removeJSComments(code).match(/(const|let)\s+kidsMeal\s*=\s*\[(?<items>.+?)\]/)
const kidsMealArray = matched.groups.items.split(",");
assert.isAtLeast(kidsMealArray.length, 1);

for (let i = 0; i < kidsMealArray.length; i++) {
  const itemMatch = kidsMealArray[i].trim().match(/^(?<quote>'|"|`)(?<meal>.+)\1$/);
  assert.isNotNull(itemMatch);
  assert.notInclude(itemMatch.groups.meal, itemMatch.groups.quote)
}
```